File: backend/src/main/java/de/ropold/backend/dto/CbcrReportResponse.java
package de.ropold.backend.dto;

import de.ropold.backend.model.CbcrReportModel;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

public record CbcrReportResponse(
        UUID id,
        CompanyBasicInfo company,
        Integer reportingYear,
        LocalDate fiscalYearEnd,
        CountryBasicInfo country,
        BigDecimal revenuesUnrelatedParty,
        BigDecimal revenuesRelatedParty,
        BigDecimal revenuesTotal,
        BigDecimal profitBeforeTax,
        BigDecimal incomeTaxPaid,
        BigDecimal incomeTaxAccrued,
        BigDecimal effectiveTaxRate,
        BigDecimal expectedTaxRate,
        BigDecimal statedCapital,
        BigDecimal accumulatedEarnings,
        BigDecimal tangibleAssets,
        BigDecimal intangibleAssets,
        Integer numberOfEmployees,
        BigDecimal revenuePerEmployee,
        String commentReference,
        String taxExplanation,
        String dataSource,
        CbcrReportModel.AuditStatus auditStatus,
        String businessActivities,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {
}

File: frontend/src/components/models/CbcrReportResponse.ts
export type CompanyBasicInfo = {
    id: string;
    name: string;
    industry: string;
    reportingCurrency: string;
}

export type CountryBasicInfo = {
    id: string;
    countryCode: string;
    countryName: string;
}

export type CbcrReportResponse = {
    id: string;
    company: CompanyBasicInfo;
    reportingYear: number;
    fiscalYearEnd?: string;
    country: CountryBasicInfo;
    revenuesUnrelatedParty?: number;
    revenuesRelatedParty?: number;
    revenuesTotal?: number;
    profitBeforeTax?: number;
    incomeTaxPaid?: number;
    incomeTaxAccrued?: number;
    effectiveTaxRate?: number;
    expectedTaxRate?: number;
    statedCapital?: number;
    accumulatedEarnings?: number;
    tangibleAssets?: number;
    intangibleAssets?: number;
    numberOfEmployees?: number;
    revenuePerEmployee?: number;
    commentReference?: string;
    taxExplanation?: string;
    dataSource: string;
    auditStatus: 'DRAFT' | 'IN_REVIEW' | 'SUBMITTED' | 'PUBLISHED' | 'FINALIZED';
    businessActivities?: string;
    createdAt: string;
    updatedAt: string;
}